src/test/java/org/opensearch/security/multitenancy/test/TenancyDefaultTenantTests.java

```diff
@@ -0,0 +1,88 @@
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.StringContains.containsString;
+
+public class TenancyDefaultTenantTests extends SingleClusterTest {
+    private final Header asAdminUser = encodeBasicHeader("admin", "admin");
+    private final Header asUser = encodeBasicHeader("kirk", "kirk");
+
+    @Override
+    protected String getResourceFolder() {
+        return "multitenancy";
+    }
+
+    @Test
+    public void testDefaultTenant_endToEndTest() throws Exception {
+        setup();
+
+        final HttpResponse getSettingResponse = nonSslRestHelper().executeGetRequest("/_plugins/_security/api/tenancy/default_tenant", asAdminUser);
+        assertThat(getSettingResponse.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertThat(getSettingResponse.findValueInJson("value"), equalTo(""));
+
+        HttpResponse getAuthinfoResponse = nonSslRestHelper().executeGetRequest("/_plugins/_security/authinfo", asAdminUser);
+        assertThat(getAuthinfoResponse.findValueInJson("default_tenant"),equalTo(""));
+
+        final HttpResponse disablePrivateTenantResponse = nonSslRestHelper().executePutRequest("/_plugins/_security/api/tenancy/private_tenant_enabled", "{\"value\":false}", asAdminUser);
+        assertThat(disablePrivateTenantResponse.getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+
+
+        final HttpResponse setPrivateTenantAsDefaultFailResponse = nonSslRestHelper().executePutRequest("/_plugins/_security/api/tenancy/default_tenant", "{\"value\": \"Private\"}", asAdminUser);
+        assertThat(setPrivateTenantAsDefaultFailResponse.getStatusCode(), equalTo(HttpStatus.SC_BAD_REQUEST));
+        assertThat(setPrivateTenantAsDefaultFailResponse.findValueInJson("error.reason"), containsString("Default tenant can not be set to Private if Private tenant is disabled."));
+
+        final HttpResponse enablePrivateTenantResponse = nonSslRestHelper().executePutRequest("/_plugins/_security/api/tenancy/private_tenant_enabled", "{\"value\":true}", asAdminUser);
+        assertThat(enablePrivateTenantResponse.getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+        final HttpResponse setPrivateTenantAsDefaultResponse = nonSslRestHelper().executePutRequest("/_plugins/_security/api/tenancy/default_tenant", "{\"value\": \"Private\"}", asAdminUser);
+        assertThat(setPrivateTenantAsDefaultResponse.getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+        final HttpResponse getSettingResponseAfterUpdate = nonSslRestHelper().executeGetRequest("/_plugins/_security/api/tenancy/default_tenant", asAdminUser);
+        assertThat(getSettingResponseAfterUpdate.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertThat(getSettingResponseAfterUpdate.findValueInJson("value"), equalTo("Private"));
+
+        getAuthinfoResponse = nonSslRestHelper().executeGetRequest("/_plugins/_security/authinfo", asAdminUser);
```

Comment:
End to end test should replicate the workflow that is used by dashboards.  If we just call the config APIs / confirm we are reading out the same value we put in it only indicates that the config was updated not that the users are seeing the correct behavior.

Same issue for the TenancyPrivateTenantEnabledTests

Reply:
Those tests will be covered in Security Dashboards plugin PR. Anyways the Dashboards also will be calling this API (DashboardsInfo) as replicated in test to get final config so that should be fine.

Reply:
We need the end to end coverage of this scenarios in the security plugin.  Otherwise breaking changes could be make in this repo.

The example from TenancyMultitenancyEnabledTests.java shows how when the feature is activated behavior changes, please use this as a starting point.  Let me know if you need help to test these scenarios.

Reply:
I have added that for TenancyPrivateTenantEnabledTests. Not sure what else to add for Default tenant tests. What test scenario are you thinking @peternied ?